### PR TITLE
Deprecates some overloaded / renamed methods

### PR DESCRIPTION
**Pull request by sabieber**

### CODE REVIEW

**Review by idlira — APPROVED** [review]

File: src/main/java/sirius/kernel/async/CallContext.java [hunk 1]
```diff
@@ -377,7 +377,9 @@ public String getLanguage() {
      * Returns the current language determined for the current thread.
      *
      * @return a two-letter language code used for the current thread.
+     * @deprecated call {@link #getLanguage()} instead.
      */
+    @Deprecated
     public String getLang() {
```
Comment (jakobvogel) [inline src/main/java/sirius/kernel/async/CallContext.java:383]:
```suggestion
    public final String getLang() {
```

File: src/main/java/sirius/kernel/async/CallContext.java [hunk 2]
```diff
@@ -413,8 +415,10 @@ public String getFallbackLanguage() {
      * Returns the current fallback language determined for the current thread.
      *
      * @return a two-letter language code used for the current thread.
+     * @deprecated call {@link #getFallbackLang()} instead.
      */
     @Nullable
+    @Deprecated
     public String getFallbackLang() {
```
Comment (jakobvogel) [inline src/main/java/sirius/kernel/async/CallContext.java:422]:
```suggestion
    public final String getFallbackLang() {
```

File: src/main/java/sirius/kernel/async/CallContext.java [hunk 3]
```diff
@@ -441,7 +445,9 @@ public void setLanguage(@Nullable String language) {
      * </p>
      *
      * @param language the two-letter language code for this thread.
+     * @deprecated call {@link #setLanguage(String)} instead.
      */
+    @Deprecated
     public void setLang(@Nullable String language) {
```
Comment (jakobvogel) [inline src/main/java/sirius/kernel/async/CallContext.java:451]:
```suggestion
    public final void setLang(@Nullable String language) {
```

File: src/main/java/sirius/kernel/async/CallContext.java [hunk 4]
```diff
@@ -467,7 +473,9 @@ public void setLanguageIfEmpty(@Nullable String language) {
      * </p>
      *
      * @param language the two-letter language code for this thread.
+     * @deprecated call {@link #setLanguageIfEmpty(String)} instead.
      */
+    @Deprecated
     public void setLangIfEmpty(@Nullable String language) {
```
Comment (jakobvogel) [inline src/main/java/sirius/kernel/async/CallContext.java:479]:
```suggestion
    public final void setLangIfEmpty(@Nullable String language) {
```

File: src/main/java/sirius/kernel/async/CallContext.java [hunk 5]
```diff
@@ -504,7 +512,9 @@ public void deferredSetLanguage(@Nonnull Consumer<CallContext> languageInstaller
      *
      * @param languageInstaller a callback which installs the appropriate language into the given call context
      *                          (is passed again to support {@link #fork() forking}).
+     * @deprecated call {@link #deferredSetLanguage(Consumer)} instead.
      */
+    @Deprecated
     public void deferredSetLang(@Nonnull Consumer<CallContext> languageInstaller) {
```
Comment (jakobvogel) [inline src/main/java/sirius/kernel/async/CallContext.java:518]:
```suggestion
    public final void deferredSetLang(@Nonnull Consumer<CallContext> languageInstaller) {
```

File: src/main/java/sirius/kernel/async/CallContext.java [hunk 6]
```diff
@@ -522,7 +532,10 @@ public void resetLanguage() {
      * Sets the lang back to <tt>null</tt>.
      * <p>
      * This method should only be used to re-initialize the language. Use {@link #setLanguage(String)} to specify a new language.
+     *
+     * @deprecated call {@link #resetLanguage()} instead.
      */
+    @Deprecated
     public void resetLang() {
```
Comment (jakobvogel) [inline src/main/java/sirius/kernel/async/CallContext.java:539]:
```suggestion
    public final void resetLang() {
```

File: src/main/java/sirius/kernel/async/CallContext.java [hunk 7]
```diff
@@ -540,7 +553,9 @@ public void setFallbackLanguage(@Nullable String fallbackLanguage) {
      * Sets the current fallback language for the current thread.
      *
      * @param fallbackLanguage the two-letter language code for this thread.
+     * @deprecated call {@link #setFallbackLanguage(String)} instead.
      */
+    @Deprecated
     public void setFallbackLang(@Nullable String fallbackLanguage) {
```
Comment (jakobvogel) [inline src/main/java/sirius/kernel/async/CallContext.java:559]:
```suggestion
    public final void setFallbackLang(@Nullable String fallbackLanguage) {
```

**Review by jakobvogel — APPROVED** [review]
I would mark the old methods final, to avoid overrides in child classes (which would end up not being called).

**Review by jakobvogel — APPROVED** [review]

**Pull request merged**